src/odemis/gui/comp/grid.py

```diff
@@ -127,19 +127,25 @@ def get_2_viewports(self):
         if [view.Shown for view in self.viewports].count(True) > 2:
             logging.warning("Found more than two viewports to show, only the first 2 are displayed in this layout.")
 
-        for v in self.viewports:
-            try:
-                if v.view is not None and v.Shown:
+        if len(self.visible_viewports) == 2:
```

Comment:
I think there is something wrong with this code. The `show_2_vert_stacked_viewports()` uses `get_2_viewports()` to *update* `.visible_viewports`. So you cannot just return `.visible_viewports`... that would be too easy.

Reply:
I actually think we can, but only **if** the number of visible viewports is 2, in another cause we need to do different stuff. That is the **else**
I have updated it according to you other comment which does simplify the reading the code.